functions/go-upload-file-s3-multipart/main.go

```diff
@@ -0,0 +1,122 @@
+package handler
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+)
+
+func Handle(w http.ResponseWriter, r *http.Request) {
+	// Entire files are sent as "multipart/form-data" so we need first to parse
+	// the request before accessing file content.
+	if err := r.ParseMultipartForm(32<<20 + 1024); err != nil {
+		panic(err)
+	}
+
+	// Save the multipartForm of the request.
+	multipartFile := r.MultipartForm.File["data"][0]
+
+	// Open the file.
+	file, err := multipartFile.Open()
+	if err != nil {
+		panic(err)
+	}
+
+	// Here we read the whole file content into fileContent as byte slice,
+	// We can directly use the file descriptor to send it to s3 to avoid saving the file locally.
+	fileContent, err := io.ReadAll(file)
+	if err != nil {
+		panic(err)
+	}
+
+	// Reach the temp dir of system to save the file
+	tmpDir := os.TempDir()
+
+	// Build the temp file path
+	tempFile := filepath.Join(tmpDir, multipartFile.Filename)
+
+	// Write file to system
+	if err := os.WriteFile(tempFile, fileContent, 0o644); err != nil {
+		panic(err)
+	}
+
+	// Always remove the file
+	defer func() {
+		os.Remove(tempFile)
+	}()
+
+	// If S3 enabled, send the file
+	if strings.ToLower(os.Getenv("S3_ENABLED")) == "true" {
+		log.Println("S3 upload enabled")
+
+		if err := connectToS3AndPushFile(tempFile); err != nil {
+			panic(err)
+		}
+	} else {
+		log.Println("S3 upload disabled")
+	}
+}
+
+// connectToS3AndPushFile will push file in parameter to the S3 bucket. Env variables (secrets)
+// must be initialized to be able to upload files.
+func connectToS3AndPushFile(filePath string) error {
+	// Read required s3 variables
+	var (
+		endpoint        = os.Getenv("S3_ENDPOINT")
+		accessKeyID     = os.Getenv("S3_ACCESSKEY")
+		secretAccessKey = os.Getenv("S3_SECRET")
+		bucketName      = os.Getenv("S3_BUCKET_NAME")
+		region          = os.Getenv("S3_REGION")
+	)
+
+	// Initialize minio client object.
+	minioClient, err := minio.New(endpoint, &minio.Options{
+		Creds:  credentials.NewStaticV4(accessKeyID, secretAccessKey, ""),
+		Secure: true,
+	})
+	if err != nil {
+		return fmt.Errorf("error opening s3 connexion : %w", err)
+	}
+
+	ctx := context.Background()
```

Comment:
I think that you could get the context from the HTTP request with `r.Context()` and pass it to this function instead of using `context.Background()`.

Reply:
done